### ISSUE TITLE
Signals enable me to implement my own ACK

### ISSUE THREAD

**Issue by muhamedkarajic** [angular/angular#53417]
### Description

Imagine that I have a global store like NgRx but where each key links to a Singal containing the state of the object. That store is global and multiple components can use the same object - you can imagine it like its firebase.

Now what I want is able to track how many components use the object, when the count go to 0 I want to wait a specific amount of time before GC-ing the signal. 

I'm not capable to implement this.

### Proposed solution

Enable me to track how many times a singal is subscribed and unsubscribed - I know this is not the right terminology but I think the point which I want to do is clear.

### Alternatives considered

None.

**Comment by JeanMeche** [comment]
Hi, I believe this is an issue you have already opened #51167 and was adressed in #50182.

**Comment by muhamedkarajic** [comment]
> Hi, I believe this is an issue you have already opened #51167 and was adressed in #50182.

You are right so if I understand correctly they just don't wanna allow this possibility cause its not what Singlas are in the Angular team oppinion deseigned for?

**Comment by pkozlowski-opensource** [comment]
> You are right so if I understand correctly they just don't wanna allow this possibility cause its not what Singlas are in the Angular team oppinion deseigned for?

Correct. Observing the act of subscribing / unsubscribing would make the signal read side-effectfull introducing the notion similar to "hot" vs. "cold" observables. This is not what signals are designed for. You might want to use a different reactive primitive for your use-cases.

**Comment by muhamedkarajic** [comment]
> > You are right so if I understand correctly they just don't wanna allow this possibility cause its not what Singlas are in the Angular team oppinion deseigned for?
> 
> Correct. Observing the act of subscribing / unsubscribing would make the signal read side-effectfull introducing the notion similar to "hot" vs. "cold" observables. This is not what signals are designed for. You might want to use a different reactive primitive for your use-cases.

I was just trying to avoid it cause for me it means I have to have a Subject and then on top of that for each component I have to cast it into a signal, its at least 2*N for memory or even M*N. I guess I have then no choice.
